[CN-536] Add ReplicatedMap CR documentation (#56)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/replicatedmap-configuration.adoc

= Configuring ReplicatedMaps
:description: pass:q[A Replicated Map is a distributed key-value data structure where the data is replicated to all members in the cluster. It provides full replication of entries to all members for high speed access. You can create and configure ReplicatedMaps using the `ReplicatedMap` custom resource.]

{description}

== Configuring a ReplicatedMap Resource

You can use any of the following configuration options in the `ReplicatedMap` resource. For more information, see the xref:api-ref.adoc#replicatedmapspec[API Reference].

[cols="20%m,80%a"]
|===
|Field|Description

|name
|Name of the ReplicatedMap configuration that you're creating. If empty, the custom resource (CR) name is used.

|hazelcastResourceName
|Defines the name of the Hazelcast resource.

|inMemoryFormat
|Internal storage format. Use either OBJECT or BINARY.

|asyncFillup

> |Specifies whether the ReplicatedMap is available for reads before the initial replication is completed. Its default value is true. If set to false, i.e., synchronous initial fill up, no exception is thrown when the ReplicatedMap is not yet ready, but null values can be seen until the initial replication is completed.

Reviewer: ```suggestion
|Specifies whether the ReplicatedMap is available for reads before the initial replication is completed. The default value is true. If set to false, a synchronous initial fill up the ReplicatedMap is allowed, and no exceptions are thrown until the initial replication is complete. Null values may be returned while the initial replication is completed.
```
I think this is accurate but please do check the suggested changes.

Author: After your comment I checked the replicated map doc page and I think the description for `async-fillup` field is wrong and I created a github issue (https://github.com/hazelcast/hz-docs/issues/568). Could you take a look at it?

Reviewer: Hi there, I've updated the description above. If you're happy that it is accurate, I'll update the replicated map page with the same description. Thanks for spotting this.

Reviewer: Hi @cagric0. I've updated the description above. If this is OK, I'll add the updated description to the Replicated Map page. Thanks again for spotting this.

Reviewer: Yes, the more direct the better. I've updated the description to match this.
